cmd/os-reliable.go

```diff
@@ -22,6 +22,50 @@ import (
 	"path"
 )
 
+// Wrapper functions to os.RemoveAll, which calls reliableRemoveAll
+// this is to ensure that if there is a racy parent directory
+// create in between we can simply retry the operation.
+func removeAll(dirPath string) (err error) {
+	if dirPath == "" {
+		return errInvalidArgument
+	}
+
+	if err = checkPathLength(dirPath); err != nil {
+		return err
+	}
+
+	if err = reliableRemoveAll(dirPath); err != nil {
```

Comment:
Based on #6208, should this be changed to
```
if err = reliableRemoveAll(dirPath); err != nil {
	switch {
		case isSysErrNotDir(err):
			return errFileAccessDenied
		case isSysErrPathNotFound(err):
			// This is a special case should be handled only for
			// windows, because windows API does not return "not a
			// directory" error message. Handle this specifically here.
			return errFileAccessDenied
		default:
			return err
	}
}
return nil
```